### PR TITLE
Add find command

### LINKED ISSUE #27: Find any contact entries (i.e OR search)
Finds persons that match any of the given fields.

Format: find [n/NAME]…​ [i/INTERNSHIP]…​ [m/MODULES]…​ [c/CCA]…​ [edu/EDUCATION]…​

- The search is case-insensitive. e.g hans will match Hans
- At least one of the optional fields must be provided.
- Only full words will be matched e.g. Han will not match Hans
- Persons matching at least one of the field will be returned (i.e. OR search). e.g. n/Hans m/cs2040s will return Hans, Bo Yang (i.e. Bo Yang is tagged with cs2040s)

Examples:

- find n/John returns john
- find i/Shopee m/cs2040s m/cs2030s returns Alex Yeoh (i.e Alex Yeoh is tagged with Shopee), David Li (i.e. David Li is tagged with cs2040s, cs2030s)

result for find iShopee mcs2040s cs2030s

### LINKED ISSUE #28: Find specific contact entries (i.e. AND search)
Finds persons that match all given fields.

Format: find -s [n/NAME]…​ [i/INTERNSHIP]…​ [m/MODULES]…​ [c/CCA]…​ [edu/EDUCATION]…​

- The search is case-insensitive. e.g hans will match Hans
- At least one of the optional fields must be provided.
- Only full words will be matched e.g. Han will not match Hans
- Only persons matching all fields will be returned (i.e. AND search). e.g. n/Bo Yang m/cs2040s will return Bo Yang (i.e. Bo Yang is tagged with cs2040s)

Examples:

- find -s n/John Doe returns John Doe
- find -s n/David Li m/cs2040s m/cs2030s returns David Li (i.e. David Li is tagged with cs2040s, cs2030s)

result for find s nDavid Li mcs2040s cs2030s

### PR DESCRIPTION
**Pull request by manu2002g**
Added find and find -s functionality. New test cases need to be added to cover the new features, as well as support for searching through tags.

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/FindAndPredicateParser.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+        personDescriptor.getStringNames().ifPresent(names ->
+                predicateList.add(new NameContainsKeywordsPredicate(names)));
+        personDescriptor.getStringPhones().ifPresent(phones ->
+                predicateList.add(new PhoneContainsKeywordsPredicate(phones)));
+        personDescriptor.getStringEmails().ifPresent(emails ->
+                predicateList.add(new EmailContainsKeywordsPredicate(emails)));
+        personDescriptor.getStringAddresses().ifPresent(list ->
+                predicateList.add(new AddressContainsKeywordsPredicate(list)));
+        Predicate<Person> predicate = predicateList.stream().reduce(x->true, Predicate::and);
```
Comment (ongkimlai) [inline src/main/java/seedu/address/logic/parser/FindAndPredicateParser.java:35]: Small nitpick, missing spaces in this line, around "->". Just to be consistent with the previous lines.

File: src/main/java/seedu/address/logic/parser/FindOrPredicateParser.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+        personDescriptor.getStringNames().ifPresent(names ->
+                predicateList.add(new NameContainsKeywordsPredicate(names)));
+        personDescriptor.getStringPhones().ifPresent(phones ->
+                predicateList.add(new PhoneContainsKeywordsPredicate(phones)));
+        personDescriptor.getStringEmails().ifPresent(emails ->
+                predicateList.add(new EmailContainsKeywordsPredicate(emails)));
+        personDescriptor.getStringAddresses().ifPresent(list ->
+                predicateList.add(new AddressContainsKeywordsPredicate(list)));
+        Predicate<Person> predicate = predicateList.stream().reduce(x->false, Predicate::or);
```
Comment (ongkimlai) [inline src/main/java/seedu/address/logic/parser/FindOrPredicateParser.java:37]: Here too

**Review by ongkimlai — APPROVED** [review]
Great job! Just have to add the whitespaces around the 2 lambda functions to be consistent with the other lines. 
LGTM 👍

File: src/main/java/seedu/address/logic/parser/FindAndPredicateParser.java (hunk 1 above)
Comment (manu2002g) [reply src/main/java/seedu/address/logic/parser/FindAndPredicateParser.java:35]: Oh good catch! Will fix in my next commit :)

**Pull request merged**